url2bc31e3489_ramonli_RabbitMQResearch_tgz-pJ8-com_mpos_lottery_te_thirdpartyservice_amqp_protobuf_CancellationMessagePulisherJ8/src/cluster/Worker.java

```diff
@@ -0,0 +1,49 @@
+package cluster;
+
+import com.rabbitmq.client.Address;
+import com.rabbitmq.client.ConnectionFactory;
+import com.rabbitmq.client.Connection;
+import com.rabbitmq.client.Channel;
+import com.rabbitmq.client.QueueingConsumer;
+
+public class Worker {
+    private final static String QUEUE_NAME = "hello";
+
+    public static void main(String[] argv) throws java.io.IOException, java.lang.InterruptedException {
+
+        ConnectionFactory factory = new ConnectionFactory();
+        /**
+         * Be aware that Sender will connect to [localhost, 5672], while the
+         * receiver connect to [localhost, 5673]. No matter which node of the
+         * cluster the client connect to, they the see same picture.
+         */
+        Connection connection = factory.newConnection(new Address[] { new Address("localhost", 5673) });
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *connection*. Currently, there are execution paths that do not contain closure statements, for example, when Channel.queueDeclare throws an exception. Ensure *connection* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).